File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        String message = "";
        if (logLine.startsWith("[INFO]")) {
            message = logLine.substring(7);
        }
        if (logLine.startsWith("[ERROR]")) {
            message = logLine.substring(8);
        }
        if (logLine.startsWith("[WARNING]")) {
            message = logLine.substring(10);
        }
        return message.strip();
    }

    public static String logLevel(String logLine) {
        String res = "";
        if (logLine.startsWith("[INFO]")) {
            res = "info";
        }
        if (logLine.startsWith("[ERROR]")) {
            res = "error";
        }
        if (logLine.startsWith("[WARNING]")) {
            res = "warning";
        }
        return res;
    }

    public static String reformat(String logLine) {
        String message = message(logLine);
        String level = logLevel(logLine);
        return String.format("%s (%s)", message, level);
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
std::string message(std::string line) {
    // return the message
    int start_idx = line.find(" ") + 1;
    return line.substr(start_idx);
}

std::string log_level(std::string line) {
    // return the log level
    int end_idx = line.find("]") - 1;
    return line.substr(1, end_idx);
}

std::string reformat(std::string line) {
    // return the reformatted message
    std::string log_msg = message(line);
    std::string log_lvl = log_level(line);

    return log_msg + " (" + log_lvl + ")";
}
}  // namespace log_line
